From d95634f7b0a4e14cd80addbdc4cd6a5eb818f02b Mon Sep 17 00:00:00 2001
From: Tommy Huang <tommy_huang@aspeedtech.com>
Date: Thu, 25 Jun 2026 08:13:34 +0800
Subject: [PATCH 2/2] i2c: ast2600: add delay to satisfy tBUF requirement

According to the I2C tBUF timing requirement, add a 10 us delay at
the end of each I2C transfer. This delay is intended for users with
more compact transfer scenarios, where the required bus free time
might not otherwise be met.

The maximum required tBUF is 4.7 us, so a 10 us delay provides
sufficient margin.

Upstream-Status: Backport [ASPEED SDK v11.03]
Signed-off-by: Tommy Huang <tommy_huang@aspeedtech.com>
Change-Id: Icac5dae56b34ceebfb19b3c7e9a81ae831333aab
Signed-off-by: rupeshs <rupeshs@ami.com>
---
 drivers/i2c/ast2600_i2c.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/drivers/i2c/ast2600_i2c.c b/drivers/i2c/ast2600_i2c.c
index 355e1fa9c9..1f3f4405c5 100644
--- a/drivers/i2c/ast2600_i2c.c
+++ b/drivers/i2c/ast2600_i2c.c
@@ -86,6 +86,9 @@ static int ast2700_i2c_read_data(struct ast2600_i2c_priv *priv, u8 chip_addr,
 		cmd = I2CM_PKT_EN;
 	}
 
+	/* add for tbuf wait */
+	udelay(10);
+
 	return ret;
 }
 
@@ -157,6 +160,9 @@ static int ast2700_i2c_write_data(struct ast2600_i2c_priv *priv, u8 chip_addr,
 		}
 	}
 
+	/* add for tbuf wait */
+	udelay(10);
+
 	return ret;
 }
 
@@ -195,6 +201,9 @@ static int ast2600_i2c_read_data(struct ast2600_i2c_priv *priv, u8 chip_addr,
 			return -EREMOTEIO;
 	}
 
+	/* add for tbuf wait */
+	udelay(10);
+
 	return 0;
 }
 
@@ -245,6 +254,9 @@ static int ast2600_i2c_write_data(struct ast2600_i2c_priv *priv, u8 chip_addr,
 			return -EREMOTEIO;
 	}
 
+	/* add for tbuf wait */
+	udelay(10);
+
 	return 0;
 }
 
-- 
2.43.0

